### DISCUSSION TITLE
Conflicting SLS IDS

### DISCUSSION THREAD

**Discussion by sjdwhiting** [Security-Onion-Solutions/securityonion#7646] [Unsupported Versions] [answered]
Hello, I've been having trouble hunting down some conflicting SLS IDS. I have had this issue in the past and been able to remedy it by removing them from the appropriate pillar files. This time however I'm having trouble locating the appropriate files. I have been having this issue since about version 2.3.100. I upgraded to the most recent release hoping it would fix itself (it did not). The below error appears when running "sudo salt-call state.highstate"

`local:
    Data failed to compile:
----------
    Detected conflicting IDs, SLS IDs need to be globally unique.
    The conflicting ID is 'sosyncusers' and is found in SLS 'base:soc' and SLS 'base:manager.sync_es_users'
----------
    Rendering SLS 'base:elasticsearch' failed: Jinja error: 'NoneType' object is not iterable
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/salt/utils/templates.py", line 502, in render_jinja_tmpl
    output = template.render(**decoded_context)
  File "/usr/lib/python3/dist-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/usr/lib/python3/dist-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python3/dist-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "<template>", line 146, in top-level template code
TypeError: 'NoneType' object is not iterable

; line 146

---
[...]
    - user: 930
    - group: 939
    - template: jinja

#sync templates to /opt/so/conf/elasticsearch/templates
{% for TEMPLATE in TEMPLATES %}    <======================
es_template_{{TEMPLATE.split('.')[0] | replace("/","_") }}:
  file.managed:
    - source: salt://elasticsearch/templates/{{TEMPLATE}}
    {% if 'jinja' in TEMPLATE.split('.')[-1] %}
    - name: /opt/so/conf/elasticsearch/templates/{{TEMPLATE.split('/')[1] | replace(".jinja", "")}}
[...]
---`

**Comment by dougburks** [comment, marked as answer]
Have you tried searching for `sosyncusers` in `/opt/so/`?
```
sudo grep -r sosyncusers /opt/so/
```

**Reply by sjdwhiting** [reply]
Results of the search:

`/opt/so/saltstack/default/salt/manager/sync_es_users.sls:sosyncusers:`
`/opt/so/saltstack/local/salt/soc/init.sls:sosyncusers:`

I removed the` init.sls` file from` /opt/so/saltstack/local/salt/soc/` so that it would go the default version. I haven't made any alterations so no concern there. 

The second part of the error I posted persisted:
 
`Rendering SLS 'base:elasticsearch' failed: Jinja error: 'NoneType' object is not iterable`

I removed the `init.sls` file from `/opt/so/saltstack/local/salt/elasticsearch` as well which seemed to fix it. This seems to have been an issue previously as well because I had a `init.sls.backup` that I had created in that directory. 

I was then able to go through and start all the containers again. I performed a `salt-call state.highstate` to verify no issues. Did a reboot for good measure to verify everything started up fine on its own. 

All is well now. 

Thanks!
